### ISSUE TITLE
Invalid credentials always sending 403 error

### ISSUE THREAD

**Issue by ahmedrafayat** [getarrays/userservice#1]
The problem is, when I try to authenticate via the `/api/login` route with invalid credentials, a 403 forbidden error response is always returned with no response body.

I even tried implementing this in CustomAuthorizationFilter, but the result is the same

```java
@Override
    protected void unsuccessfulAuthentication(HttpServletRequest request, HttpServletResponse response, AuthenticationException failed) throws IOException, ServletException {
        response.sendError(401, "Bad Credentials");
    }
```

**Comment by getarrays** [comment]
Please take the code and compare it to yours to see if there is something you're missing.

**Comment by ahmedrafayat** [comment]
I dont think I missed anything. Anyway, I ran the project directly and can reproduce the same issue still.

**Comment by davidnjau** [comment]
I'm also experiencing the same issue.

**Comment by Abdqadr1** [comment]
> I dont think I missed anything. Anyway, I ran the project directly and can reproduce the same issue still.

Yeah me too. I think the new version of spring is the problem. I was using 2.6.3 when I encountered the problem but when I changed it to 2.5.2 as it is in this project. Everything worked fine

**Comment by mbedru** [comment]
happened the same to me but my solution was different
in SecurityConfig  ==> this line 
        http.authorizeRequests().antMatchers("/api/login/**", "/api/token/refresh/**").permitAll();

i forgot putting slash"/" before api 
like this ==>   "api/token/refresh/**" instead of  "/api/token/refresh/**"

this means the permitAll function contains the /api/login but not /api/token/refresh

just check every line in SecurityConfig.java file
